security/openpgp.ksy

```diff
@@ -0,0 +1,503 @@
+doc-ref: https://tools.ietf.org/html/rfc4880
+seq:
+ - id: packets
+   type: packet
+   repeat: eos
+types:
+  packet:
+    -webide-representation: '{packet_type_old}'
+    seq:
+      - id: one
+        type: b1
+      - id: new_packet_format
+        type: b1
+      - id: packet_type_new
+        type: b6
+        if: new_packet_format
+        enum: packet_tags
+      - id: packet_type_old
+        type: b4
+        if: not new_packet_format
+        enum: packet_tags
+      - id: length_type
+        type: b2
+        if: not new_packet_format
+      - id: body
+        type: 
+          switch-on: new_packet_format
+          cases:
+            #true: new_packet
+            false: old_packet
+            
+  old_packet:
+    seq:
+      - id: length
+        type: 
+          switch-on: _parent.length_type
+          cases:
+            0: u1
+            1: u2
+            2: u4
+      - id: body
+        size: length
+        type:
+          switch-on: _parent.packet_type_old
+          cases:
+            packet_tags::user_id_packet: user_id_packet
+            packet_tags::signature_packet: signature_packet
+            packet_tags::public_key_packet: public_key_packet
+            packet_tags::public_subkey_packet: public_key_packet
+            packet_tags::secret_key_packet: secret_key_packet
+            packet_tags::secret_subkey_packet: public_key_packet
+  
+  public_key_packet:
+    seq:
+      - id: version
+        type: u1
+      - id: timestamp
+        type: u4
+      - id: public_key_algorithm
+        type: u1
+        enum: public_key_algorithms
+      - id: alg_length
+        type: u2
+      - id: rsa_n
+        size: alg_length / 8
+      - id: padding
+        type: u2
+      - id: rsa_e
+        size: 3
+      
+  user_id_packet:
+    seq:
+      - id: user_id
+        size-eos: true
+        type: str
+
+  signature_packet:
+    seq: 
+      - id: version
+        type: u1
+        # enum: TODO switch?
+      - id: signature_type
+        type: u1
+        # enum: TODO 5.2.1
+      - id: public_key_algorithm
+        type: u1
+        enum: public_key_algorithms
+      - id: hash_algorithm
+        type: u1
+        enum: hash_algorithms
+      - id: hashed_subpacket_length
+        type: u2
+      - id: hashed_subpackets
+        type: subpackets
+        size: hashed_subpacket_length
+      - id: unhashed_subpacket_length
+        type: u2
+      - id: unhashed_subpackets
+        type: subpackets
+        size: unhashed_subpacket_length
+      - id: left_signed_hash
+        type: u2
+      - id: rsa_n
+        type: u2
+      - id: signature
+        size-eos: true
+    
+  secret_key_packet:
+    seq:
+      - id: public_key
+        type: public_key_packet
+      - id: string_to_key
+        type: u1
+      - id: symmetric_encryption_algorithm
+        type: u1
+        enum: symmetric_key_algorithm
+        if: string_to_key >= 254
+      - id: more
+        size-eos: true
```

Comment:
I believe this is just padding, isn't it? If that's it, then it's not really needed, I guess.

Reply:
Actually `more` is the secret key. I fixed it.